edgelet/kube-client/src/client.rs

```diff
@@ -81,38 +81,44 @@ where
         namespace: &str,
         config_map: &api_core::ConfigMap,
     ) -> impl Future<Item = api_core::ConfigMap, Error = Error> {
-        api_core::ConfigMap::create_core_v1_namespaced_config_map(namespace, config_map, None)
-            .map_err(Error::from)
-            .map(|req| {
-                self.request(req).and_then(|response| match response {
-                    api_core::CreateCoreV1NamespacedConfigMapResponse::Ok(config_map)
-                    | api_core::CreateCoreV1NamespacedConfigMapResponse::Created(config_map)
-                    | api_core::CreateCoreV1NamespacedConfigMapResponse::Accepted(config_map) => {
-                        Ok(config_map)
-                    }
-                    err => {
-                        debug!("Create config map failed with {:#?}", err);
-                        Err(Error::from(ErrorKind::Response))
-                    }
-                })
+        api_core::ConfigMap::create_namespaced_config_map(
+            namespace,
+            config_map,
+            api_core::CreateNamespacedConfigMapOptional::default(),
+        )
+        .map_err(Error::from)
+        .map(|(req, _)| {
```

Comment:
You should thread the `response_body` return value from the request functions to your `edgelet_kube::Client` instead of ignoring it. Then you get type-safety for the response type.

See https://github.com/Arnavion/k8s-openapi/blob/master/k8s-openapi-tests/src/lib.rs#L240-L246 for an example.

Reply:
Fixed